File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        ArrayList<Integer> res = new ArrayList<>();
        Stack<TreeNode> stack = new Stack<>();
        TreeNode head = root;
        while(head != null || !stack.isEmpty()){
            
            while(head != null){
                stack.push(head);
                head = head.left;
            
            }
            head = stack.pop();
            res.add(head.val);
            head = head.right;
            
            }
        return res;
        }
        
    }

File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def helper(self, root, res):
        if root != None:
            if root.left != None:
                self.helper(root.left, res)
            res.append(root.val)
            if root.right != None:
                self.helper(root.right, res)
            
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
            res = [] 
            self.helper(root,res)
            return res